pkg/solana/chainreader/chain_reader_test.go

```diff
@@ -621,6 +624,11 @@ func (r *wrappedTestChainReader) GetLatestValue(ctx context.Context, contractNam
 
 		fallthrough
 	default:
+		// If you called a method and the service is not started
+		if err := r.service.Ready(); err != nil {
+			return fmt.Errorf("service not ready. err: %w", err)
+		}
+
```

Comment:
Shouldn't this check be before the switch statement?

Reply:
Good catch! Fixed